polynote-frontend/polynote/util/helpers.ts

```diff
@@ -60,6 +68,9 @@ export function changedKeys<T extends Record<string, any>>(oldT: T, newT: T): (k
 
 export function deepFreeze<T>(obj: T) {
     function go(obj: T) {
+        if (obj instanceof StateHandler) {
```

Comment:
👍  good idea!

Reply:
Maybe we could also test for class instances which break equality? (or at least objects with anonymous functions like we talked about)

Reply:
Let's follow up with that in another PR.

Reply:
For sure! 👍